File: java-user-management-service/src/main/java/com/ghcp/starter/user_management/dto/UserResponse.java
package com.ghcp.starter.user_management.dto;

import lombok.Builder;
import lombok.Data;
import java.time.LocalDateTime;
import java.util.UUID;

@Data
@Builder
public class UserResponse {
    private UUID id;
    private String email;
    private String role;
    private LocalDateTime createdAt;
    private Profile profile;

    @Data
    @Builder
    public static class Profile {
        private String name;
        private String address;
        private String phone;
        private String avatarUrl;
        private String dateOfBirth;
    }
}

File: python-user-management-service/user_response.py
# GH Copilot code - starts
from user import User
from user_profile import UserProfile

class UserResponse:
    def __init__(self, id, email, role, created_at, profile):
        self.id = id
        self.email = email
        self.role = role
        self.created_at = created_at
        self.profile = profile

    @classmethod
    def from_user(cls, user: User):
        profile_dict = {
            "name": user.profile.name,
            "address": user.profile.address,
            "phone": user.profile.phone,
            "avatarUrl": getattr(user.profile, "avatar_url", None),
            "dateOfBirth": getattr(user.profile, "date_of_birth", None)
        }
        return cls(
            id=user.id,
            email=user.email,
            role=user.role,
            created_at=user.created_at.isoformat() + "Z",
            profile=profile_dict
        )
# GH Copilot code - end
